pkg/networkservice/common/interpose/server.go

```diff
@@ -93,15 +97,15 @@ func (l *interposeServer) Request(ctx context.Context, request *networkservice.N
 		// Iterate over all cross connect NSEs to check one with passed state.
 
 		l.endpoints.Range(func(key string, crossNSEURL *url.URL) bool {
-			crossCTX := clienturlctx.WithClientURL(ctx, crossNSEURL)
+			crossCTX := extend.WithValuesFromContext(ctx, clienturlctx.WithClientURL(context.Background(), crossNSEURL))
```

Comment:
Did we loose a timeout here?

Reply:
Please look at `extend`, we only lose values here.

Reply:
ok